overlord/configstate/configcore/watchdog.go

```diff
@@ -0,0 +1,142 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+var watchdogConfigKeys = map[string]bool{
+	"RuntimeWatchdogSec":  true,
+	"ShutdownWatchdogSec": true,
+}
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations["core.watchdog.runtime-timeout"] = true
+	supportedConfigurations["core.watchdog.shutdown-timeout"] = true
+}
+
+func watchdogConfPath() string {
+	return filepath.Join(dirs.GlobalRootDir, "/etc/systemd/system.conf.d/")
```

Comment:
If SnapServicesDir is not appropriate, let's please have SnapSystemdConfDir for that one. It should really be defined there rather than here.

That means we don't need this function anymore as well.

Reply:
Done